utils/tinymix.c

```diff
@@ -323,54 +375,64 @@ static void tinymix_set_byte_ctl(struct mixer_ctl *ctl,
 
 static int is_int(const char *value)
 {
-    return (value[0] >= '0') || (value[0] <= '9');
+    return value[0] >= '0' || value[0] <= '9';
 }
 
 struct parsed_int
 {
-  /** Wether or not the integer was valid. */
-  int valid;
-  /** The value of the parsed integer. */
-  int value;
-  /** The number of characters that were parsed. */
-  unsigned int length;
-  /** The number of characters remaining in the string. */
-  unsigned int remaining_length;
-  /** The remaining characters (or suffix) of the integer. */
-  const char* remaining;
+    /** Wether or not the integer was valid. */
+    int valid;
+    /** The value of the parsed integer. */
+    int value;
+    /** The number of characters that were parsed. */
+    unsigned int length;
+    /** The number of characters remaining in the string. */
+    unsigned int remaining_length;
+    /** The remaining characters (or suffix) of the integer. */
+    const char* remaining;
 };
 
 static struct parsed_int parse_int(const char* str)
 {
-  struct parsed_int out = {
-    0 /* valid */,
-    0 /* value */,
-    0 /* length */,
-    0 /* remaining length */,
-    "" /* remaining characters */
-  };
+    struct parsed_int out = {
+        0 /* valid */,
+        0 /* value */,
+        0 /* length */,
+        0 /* remaining length */,
+        NULL /* remaining characters */
+    };
 
-  unsigned int max = strlen(str);
+    size_t length = strlen(str);
+    size_t i = 0;
+    int negative = 0;
 
-  for (unsigned int i = 0; i < max; i++) {
+    if (i < length && str[i] == '-') {
+        negative = 1;
+        i++;
+    }
 
-    char c = str[i];
+    while (i < length) {
+        char c = str[i++];
```

Comment:
This probably isn't a problem at the moment, but this abbreviated form changes the way the loop works. When `c` is not a digit, the loop should break without incrementing `i`. This change will cause `i` to point to `1` passed the first non-digit character, instead of leaving it at the character that did not match. You can just add a `i--` before the break statement or switch back to a for loop.

Reply:
Thanks for the reviewing. I will correct it.